tests/ssg_test_suite/xml_operations.py

```diff
@@ -80,21 +80,39 @@ def datastream_root(ds_location, save_location=None):
             tree.write(save_location)
 
 
-def remove_machine_platform(root):
-    remove_machine_only_from_element(root, "xccdf-1.2:Rule")
-    remove_machine_only_from_element(root, "xccdf-1.2:Group")
-
-
-def remove_machine_only_from_element(root, element_spec):
+def remove_platforms_from_element(root, element_spec, platforms):
     query = BENCHMARK_QUERY + "//{0}".format(element_spec)
     elements = root.findall(query, PREFIX_TO_NS)
     for el in elements:
         platforms = el.findall("./xccdf-1.2:platform", PREFIX_TO_NS)
         for p in platforms:
```

Comment:
```suggestion
        platforms_xml = el.findall("./xccdf-1.2:platform", PREFIX_TO_NS)
        for p in platforms_xml:
```

it's conflicting with the new `platforms` you have just created.